### PR TITLE
[DataGrid] Remove unnecessary syntax in JSDoc

### LINKED ISSUE #10568: [core] Remove unnecessary syntax from JS Docs
### Related page

JS Documentation

### Kind of issue

Other

### Issue description

I've found that for no reason ( I guess ) in JS Docs there were couple of times at which after the `@default` keyword there was a colon symbol which isn't required by syntax of JS docs and has absolutely no purpose ( additionally it looks ugly ). so it should be removed I guess

### Context 🔦

Screenshot 2023-10-11 at 00 21 35

**Search keywords**: Docs, docs syntax

### PR DESCRIPTION
**Pull request by Lev-Shapiro**
This PR fixes #10568 

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

**Comment by LukasTy** [comment]
Hello @Lev-Shapiro 
Did you intend to close this PR or was it an accident? 🤔

**Comment by Lev-Shapiro** [comment]
An accident, but could you help me understand what it means when `ci/circleci` test fails? I've only changed comments so I have no idea how could it change something that will give any error

**Comment by LukasTy** [comment]
> An accident, but could you help me understand what it means when `ci/circleci` test fails? I've only changed comments so I have no idea how could it change something that will give any error

Gotcha. 👌 
Well, you have updated JSDoc for props and we use that to generate `PropTypes` as well as our API documentation.
https://github.com/mui/material-ui/blob/HEAD/CONTRIBUTING.md#updating-the-component-api-documentation
Please run `yarn docs:api` locally and commit the resulting diff to avoid the error. 😉

File: packages/grid/x-data-grid-pro/src/models/gridGroupingColDefOverride.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ export interface GridGroupingColDefOverride<R extends GridValidRowModel = any>
    * The field from which we want to apply the sorting and the filtering for the grouping column.
    * It is only useful when `props.rowGroupingColumnMode === "multiple"` to decide which grouping criteria should be used for sorting and filtering.
    * Do not have any effect when building the tree with the `props.treeData` feature.
-   * @default: The sorting and filtering is applied based on the leaf field in any, otherwise based on top level grouping criteria.
+   * @default The sorting and filtering is applied based on the leaf field in any, otherwise based on top level grouping criteria.
```
Comment (michelengelen) [inline packages/grid/x-data-grid-pro/src/models/gridGroupingColDefOverride.ts:18]: this is not the intended usage of the `@default` directive. @cherniavskii do we use it that way in the codebase very often?

Comment (Lev-Shapiro) [reply packages/grid/x-data-grid-pro/src/models/gridGroupingColDefOverride.ts:18]: I guess I could also answer this question. I've found yesterday many incorrect usages of the @default directive some have syntax problems and some incorrect usages of that directive, I could try make a more global PR and an issue to fix it, sounds good?

Comment (LukasTy) [reply packages/grid/x-data-grid-pro/src/models/gridGroupingColDefOverride.ts:18]: It's used in a similar way in the Pickers codebase a few times as well. 🤔 
Maybe it's not the most correct usage, but it works and helps us to build the API documentation that would be useful. 🙈

Comment (michelengelen) [reply packages/grid/x-data-grid-pro/src/models/gridGroupingColDefOverride.ts:18]: @Lev-Shapiro thanks, but that's not what I intended with that comment. I know it is not the correct usage of it, but my intention was to know if we use it that way elsewhere or if it is the only time.

As mentioned by @LukasTy it does not pose any threat and if it is a valuable tool we should continue using it like this! 💯

File: packages/grid/x-data-grid-premium/src/DataGridPremium/DataGridPremium.tsx [hunk 2]
```diff
@@ -1008,7 +1008,7 @@ DataGridPremiumRaw.propTypes = {
   /**
    * If `true`, the grid will not use `valueFormatter` when exporting to CSV or copying to clipboard.
    * If an object is provided, you can choose to ignore the `valueFormatter` for CSV export or clipboard export.
-   * @default: false
+   * @default false
```
Comment (LukasTy) [inline packages/grid/x-data-grid-premium/src/DataGridPremium/DataGridPremium.tsx:1011]: This is technically touching non-MIT code.
Do we need to go with the SLA process for even non-functional changes?
cc @joserodolfofreitas @DanailH

Comment (cherniavskii) [reply packages/grid/x-data-grid-premium/src/DataGridPremium/DataGridPremium.tsx:1011]: @LukasTy I believe we need a CLA for *any* change in the commercially licensed folder.

Comment (cherniavskii) [reply packages/grid/x-data-grid-premium/src/DataGridPremium/DataGridPremium.tsx:1011]: Although in our [CLA docs](https://www.notion.so/mui-org/CLA-Contributor-License-Agreement-92ece655b1584b10b00e4de9e67eedb0#50f662641b8048cb969f1ad400d04983) it says:

> If the change is nontrivial and user-facing, the CLA needs to be signed.

Comment (oliviertassinari) [reply packages/grid/x-data-grid-premium/src/DataGridPremium/DataGridPremium.tsx:1011]: I have updated https://www.notion.so/mui-org/CLA-Contributor-License-Agreement-92ece655b1584b10b00e4de9e67eedb0?pvs=4#a1f27812396a4cfd9a7891e61fb2acd7 to detail this more, inspired by OpenSSL. Here, I think we can consider it's a trivial change, if the author agrees too, we could move forward without CLA

File: packages/grid/x-data-grid-generator/src/services/tree-data-generator.ts [hunk 3]
```diff
@@ -11,13 +11,13 @@ export interface AddPathToDemoDataOptions {
 
   /**
    * The depth of the tree
-   * @default: 1
+   * @default 1
    */
   maxDepth?: number;
```
Comment (oliviertassinari) [inline packages/grid/x-data-grid-generator/src/services/tree-data-generator.ts:17]: I wouldn't mind we fix the blank line convention on these files at the same time 😁

```suggestion
```

File: packages/grid/x-data-grid-premium/src/DataGridPremium/DataGridPremium.tsx (hunk 2 above)
Comment (Lev-Shapiro) [reply packages/grid/x-data-grid-premium/src/DataGridPremium/DataGridPremium.tsx:1011]: Yeah, sure, let's move forward

**Review by cherniavskii — APPROVED** [review]
Thanks!

**Pull request merged**